processor/src/host/mast_forest_store.rs

```diff
@@ -26,7 +26,8 @@ impl MemMastForestStore {
         let mast_forest = Arc::new(mast_forest);
 
         for root in mast_forest.procedure_roots() {
-            self.mast_forests.insert(root, mast_forest.clone());
+            let root_digest = mast_forest[*root].digest();
+            self.mast_forests.insert(root_digest, mast_forest.clone());
         }
```

Comment:
Should we create a new convenience accessor in `MastForest`? Maybe something like `MastForest::procedure_digests()`?

Reply:
done